jetty-core/jetty-util/src/main/java/org/eclipse/jetty/util/resource/FileSystemPool.java

```diff
@@ -183,6 +212,33 @@ private void unmount(URI fsUri)
         }
     }
 
+    private Path createEmptyFileWithParents(Path path) throws IOException
+    {
+        Path createdRootPath = null;
+        if (!Files.exists(path.getParent()))
+            createdRootPath = createDirWithAllParents(path.getParent());
+        Files.createFile(path);
+        if (createdRootPath == null)
+            createdRootPath = path;
+        return createdRootPath;
+    }
+
+    private Path createDirWithAllParents(Path path) throws IOException
```

Comment:
Why not `Files.createDirectories()`?

Reply:
`Files.createDirectories()` doesn't tell what directories were created, if any. We need that information to delete whatever temp dir was created after closing the filesystem.